src/bridges/compound/CompoundBridge.sol

```diff
@@ -97,13 +126,30 @@ contract CompoundBridge is BridgeBase {
                 IERC20 tokenOut = IERC20(_outputAssetA.erc20Address);
                 tokenIn.redeem(_inputValue);
                 outputValueA = tokenOut.balanceOf(address(this));
-                // Using safeIncreaseAllowance(...) instead of approve(...) here because tokenOut can be Tether
-                tokenOut.safeIncreaseAllowance(ROLLUP_PROCESSOR, outputValueA);
             } else {
                 revert ErrorLib.InvalidInputA();
             }
         } else {
             revert ErrorLib.InvalidAuxData();
         }
     }
+
+    function _preApprove(ICERC20 _cToken) private {
+        uint256 allowance = _cToken.allowance(address(this), ROLLUP_PROCESSOR);
+        if (allowance < type(uint256).max) {
+            _cToken.approve(ROLLUP_PROCESSOR, type(uint256).max - allowance);
+        }
+        if (address(_cToken) != cETH) {
+            IERC20 underlying = IERC20(_cToken.underlying());
+            // Using safeIncreaseAllowance(...) instead of approve(...) here because underlying can be Tether;
+            allowance = underlying.allowance(address(this), address(_cToken));
+            if (allowance < type(uint256).max) {
+                underlying.safeIncreaseAllowance(address(_cToken), type(uint256).max - allowance);
```

Comment:
Add a test to check the case where `0 < allowance < type(uint256).max` for all the assets. Should never happen as the allowance is set to max and we don't expect to ever reach that, but since we can hit it, should be tested.

Reply:
Good point, will add 👍